slack_block.py

```diff
@@ -0,0 +1,85 @@
+from slacker import Slacker
+
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.metadata.properties import VersionProperty, StringProperty, \
+    ExpressionProperty, ObjectProperty, PropertyHolder
+
+
+class BotInformation(PropertyHolder):
+
+    name = ExpressionProperty(title='Name', default='')
+    icon = ExpressionProperty(
+        title='Icon (URL or emoji)', default=':smiley_cat:')
+
+    def get_bot_details(self, signal):
```

Comment:
This is the first time I've seen a helper method defined in a PropertyHolder like this. Cool.